pkg/image/api/validation/validation.go

```diff
@@ -297,7 +299,38 @@ func ValidateImageStreamTagUpdate(newIST, oldIST *api.ImageStreamTag) field.Erro
 	return result
 }
 
-func ValidateImageStreamImport(isi *api.ImageStreamImport) field.ErrorList {
+func ValidateRegistryAllowedForImport(path *field.Path, name, registryHost, registryPort string, allowedRegistries *serverapi.AllowedRegistries) field.ErrorList {
+	errs := field.ErrorList{}
+	allowedRegistriesForHumans := []string{}
+	for _, registry := range *allowedRegistries {
+		allowedRegistryHost, allowedRegistryPort := "", ""
+		parts := strings.Split(registry.DomainName, ":")
+		switch len(parts) {
+		case 1:
+			allowedRegistryHost = parts[0]
+			if registry.Insecure {
+				allowedRegistryPort = "80"
+			} else {
+				allowedRegistryPort = "443"
+			}
+		case 2:
+			allowedRegistryHost, allowedRegistryPort = parts[0], parts[1]
+		default:
+			// TODO: Should we panic/glog here? The format of allowed registries should be
+			// validated during the server startup.
+			glog.V(5).Infof("invalid registry domain format: %s", registry.DomainName)
+			continue
+		}
+		if util.IsWildcardMatch(registryHost, allowedRegistryHost) && registryPort == allowedRegistryPort {
```

Comment:
So if the whitelist has a benevolent entry with `insecure==true` and someone wants import a from the repo securely, he won't be allowed to.

Another case this fails is an import from a registry hosted on `:443` with invalid certificate - user is forced to use the insecure flag to no avail because the ports won't match here.